crates/cxx-qt-lib/src/core/qtlogging.cpp

```diff
@@ -0,0 +1,48 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Joshua Goins <joshua.goins@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#include "cxx-qt-lib/qtlogging.h"
+
+#include "../assertion_utils.h"
+
+// QMessageLogContext has three "const char*" members for line, category, etc
+// https://codebrowser.dev/qt5/qtbase/src/corelib/global/qlogging.h.html#QMessageLogContext
+assert_alignment_and_size(QMessageLogContext, alignof(intptr_t), sizeof(intptr_t) * 4);
+
+static_assert(::std::is_trivially_copyable<QMessageLogContext>::value,
+              "QMessageLogContext must be trivially copyable");
+
+int qmessagelogcontext_line(const QMessageLogContext &context) {
```

Comment:
this file needs clang-format running on it :-)

Reply:
Done